fix: wording color well -> color picker

toqueteos opened the pull request:
### Description

I was reading this page and kept wondering what a "color well" was.

### Motivation

~Typos do never help.~

Apparently [color well is a thing][1], an [Apple thing] as Google doesn't return much more than that.

Color wheel is a much more extended term, but color picker, as suggested, is good enough.

### Additional details

N/A

### Related issues and pull requests

N/A

[1]: https://www.google.com/search?q=color+well
[2]: https://developer.apple.com/design/human-interface-guidelines/components/selection-and-input/color-wells/

toqueteos committed:
fix: wording color well -> color picker

Josh-Cena commented:
If a page uses a term consistently, there's a fair chance that it's intentional. Indeed, [color well](https://developer.apple.com/design/human-interface-guidelines/components/selection-and-input/color-wells/) is an existing concept, and it's better than "color wheel", because not all browsers' color picker implementation uses a color wheel.

hamishwillee commented:
> I think "color picker" may be less confusing for people?

This is a good idea. @toqueteos Would you be OK changing it in line with this?

toqueteos commented:
Done, I've rebased with latest `main` and changed to color picker terminology.

Josh-Cena commented on files/en-us/web/html/element/input/color/index.md:
```diff
@@ -77,7 +77,7 @@ Inputs of type `color` are simple, due to the limited number of attributes they
 ### Providing a default color
 
-You can update the simple example above to set a default value, so that the color well is pre-filled with the default color and the color picker (if any) will also default to that color:
+You can update the simple example above to set a default value, so that the color picker is pre-filled with the default color and the color picker (if any) will also default to that color:
```
"Color well" refers to the little square filled with the color. IIRC, this comes from painting, but I didn't find any info either. In any case, "color picker" here is now redundant and we should make sure the content on this page still makes sense.

schalkneethling replied:
I believe the sentence reads a little odd now because "color well" here refers to the initial UI element as seen in this screen shot. Color picker, refers to the UI that might launch (if supported by the browser) when the user clicks on the color well.

![screenshot showing the initial state of an input element of type color. It shows a rectangle filled with a red color.](https://user-images.githubusercontent.com/10350960/215471490-d14ea9d4-ea7e-49a2-9d54-94776754fd62.png)

schalkneethling commented on files/en-us/web/html/element/input/color/index.md:
```diff
@@ -124,16 +124,16 @@ Let's create an example which does a little more with the color input by trackin
 
-<label for="colorWell">Color:</label>
-<input type="color" value="#ff0000" id="colorWell" />
+<label for="colorPicker">Color:</label>
+<input type="color" value="#ff0000" id="colorPicker" />
```
@dipikabh is this naming of attribute values consistent with out style guide? I would have thought we would use kebab case, i.e.

```suggestion
<label for="color-picker">Color:</label>
<input type="color" value="#ff0000" id="color-picker" />
```

dipikabh replied:
You're right, kebab case is what we recommend: https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/HTML#class_and_id_names. We specifically say not to use camelCase.

bsmth committed:
Update files/en-us/web/html/element/input/color/index.md

Co-authored-by: Schalk Neethling <schalk.neethling@mechanical-ink.co.za>

bsmth committed:
Merge branch 'main' into patch-1

bsmth commented on files/en-us/web/html/element/input/color/index.md:
```diff
@@ -111,15 +111,15 @@ Once the page is loaded, our `load` event handler, `startup()`, is called:
-  colorWell.addEventListener("input", updateFirst, false);
-  colorWell.addEventListener("change", updateAll, false);
-  colorWell.select();
+  colorPicker = document.querySelector("#colorPicker");
```
```suggestion
  colorPicker = document.querySelector("#color-picker");
```

bsmth committed:
Apply suggestions from code review

use `color-picker` for the id names: https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/HTML#class_and_id_names

estelle committed:
Merge branch 'main' into patch-1

estelle reviewed (approved):
Thanks! 🎉

Josh-Cena commented:
Sorry I should have blocked it :/ There's still one outstanding comment here: https://github.com/mdn/content/pull/22758#discussion_r1041290941

Not a big deal though, someone may bite the bullet later when they care enough.
